pkgs/by-name/zo/zoom-us/package.nix

```diff
@@ -6,19 +6,49 @@
   xdgDesktopPortalSupport ? true,
   callPackage,
   buildFHSEnv,
+  gzip,
+  xar,
+  cpio,
 }:
 
 let
-  unpacked = stdenv.mkDerivation (finalAttrs: {
-    pname = "zoom";
-    version = "6.4.6.1370";
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  # Zoom versions are released at different times for each platform and often
+  # with different versions. We write them on three lines like this (rather
+  # than using {}) so that the updater script can find where to edit them.
+  versions.aarch64-darwin = "6.4.6.53970";
+  versions.x86_64-darwin = "6.4.6.53970";
+  versions.x86_64-linux = "6.4.6.1370";
 
-    src = fetchurl {
-      url = "https://zoom.us/client/${finalAttrs.version}/zoom_x86_64.pkg.tar.xz";
+  srcs = {
+    x86_64-linux = fetchurl {
+      url = "https://zoom.us/client/${versions.x86_64-linux}/zoom_x86_64.pkg.tar.xz";
       hash = "sha256-Y+8garSqDcKLCVv1cTiqGEfrGKpK3UoXIq8X4E8CF+8=";
     };
+    aarch64-darwin = fetchurl {
+      url = "https://zoom.us/client/${versions.aarch64-darwin}/zoomusInstallerFull.pkg?archType=arm64";
+      name = "zoomusInstallerFull.pkg";
+      hash = "sha256-yNsiFZNte4432d8DUyDhPUOVbLul7gUdvr+3qK/Y+tk=";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://zoom.us/client/${versions.x86_64-darwin}/zoomusInstallerFull.pkg";
+      hash = "sha256-Ut93qQFFN0d58wXD5r8u0B17HbihFg3FgY3a1L8nsIA=";
+    };
+  };
 
-    dontUnpack = true;
+  unpacked = stdenv.mkDerivation (finalAttrs: {
+    pname = "zoom";
+    version = versions.${system} or throwSystem;
+
+    src = srcs.${system} or throwSystem;
+
+    dontUnpack = lib.optionals stdenv.hostPlatform.isLinux true;
+    unpackPhase = lib.optionalString stdenv.hostPlatform.isDarwin ''
+      ${lib.getExe xar} -xf $src
```

Comment:
I think it's cleaner to put those programs into `nativeBuildInputs` (with the exception of `gzip` -- see above) and then just use them from the resulting `$PATH`, i.e., without absolute paths.  It might be a theoretical concern in this case, but I suspect that -- in general -- it helps cross-"compiling", and it appears to be a common pattern in nixpkgs.

When this is done, the comment about missing libraries below probably needs adaption.